docs/18-machine-words-and-bits.markdown

```diff
@@ -0,0 +1,151 @@
+# Chapter 18. Machine Words and Bits
+
+The MDL facility for dealing with uninterpreted machine words and bits
+involves two data TYPEs: WORD and BITS. A WORD is simply an
+uninterpreted machine word, while a BITS is a "pointer" to a set of
+bits within a WORD. Operating on WORDs is usually done only when
+compiled programs are used (chapter 19).
+
+## 18.1. WORDs
+
+A `WORD` in MDL is a PDP-10 machine word of 36 bits. A `WORD` always
+`PRINT`s in "# format", and its contents are always printed in octal
+(hence preceded and followed by *). Examples:
+
+    #WORD 0                  ;"all 0s"$
+    #WORD *000000000000*
+
+    #WORD *2000*             ;"one bit 1"$
+    #WORD *000000002000*
+
+    #WORD *525252525252*     ;"every other bit 1"$
+    #WORD *525252525252*
+
+`WORD` is its own `PRIMTYPE`; it is also the `PRIMTYPE` of `FIX`,
+`FLOAT`, `CHARACTER`, and any other `TYPE` which can fit its data into
+one machine word.
+
+A `WORD` cannot be an argument to `+`, `-`, or indeed any `SUBR`s
+except for `CHTYPE`, `GETBITS`, `PUTBITS` and several bitwise logical
+functions, all to be described below. Thus any arithmetic bit
+manipulation must be done by `CHTYPE`ing a `WORD` to `FIX`, doing the
+arithmetic, and then `CHTYPE`ing back to `WORD`. However, bit
+manipulation can be done without `CHTYPE`ing the thing to be played
+with to a `WORD`, so long as it is of `PRIMTYPE` `WORD`; the result of
+the manipulation will be of the same `TYPE` as the original object or
+can be `CHTYPE`d to it.
+
+## 18.2. BITS
+
+An object of `TYPE` `BITS` is of `PRIMTYPE` `WORD`, and `PRINT`s just
+like a `WORD`. The internal form of a `BITS` is precisely that of a
+PDP-10 "byte pointer", which is, in fact, just what a `BITS` is.
+
+For purposes of explaining what a `BITS` is, assume that the bits in a
+`WORD` are numbered from **right** to **left**, with the rightmost bit
+numbered 0 and the leftmost numbered 35, as in
+
+    35 34 33 ... 2 1 0
+
+(This is not the "standard" ordering: the "standard" ordering one goes
+from left to right.)
+
+A `BITS` is most conveniently created via the `SUBR` `BITS`:
+
+    <BITS width:fix right-edge:fix>
+
+returns a `BITS` which "points to" a set of bits *width* wide. with
+rightmost bit *right-edge*. Both arguments must be of `TYPE` `FIX`,
+and the second is optional, 0 by default.
+
+Examples: the indicated application of `BITS` returns an object of
+`TYPE` `BITS` which points to the indicated set of bits in a `WORD`:
+
+    <BITS 7>         35 ... 7 6 ... 0
```

Comment:
Without the formatting to show which bits are being "indicated", the point of this example is kinda lost... maybe we could make it a table instead of a code block?